### PR TITLE
created DTOs, controller and mapper

### PR DESCRIPTION
**Pull request by JeniaSan**

### CODE REVIEW

File: src/main/java/com/polishuchenko/bookstore/config/MapperConfig.java [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package com.polishuchenko.bookstore.config;
+
+import org.mapstruct.InjectionStrategy;
+import org.mapstruct.NullValueCheckStrategy;
+
+@org.mapstruct.MapperConfig(
```
Comment (holsoni) [inline src/main/java/com/polishuchenko/bookstore/config/MapperConfig.java]: We can use MapperConfig straght and not write package name

File: src/main/java/com/polishuchenko/bookstore/controller/BookController.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RequiredArgsConstructor
+@RestController
+@RequestMapping(value = "/api/books")
+public class BookController {
+
```
Comment (holsoni) [inline src/main/java/com/polishuchenko/bookstore/controller/BookController.java]:
```suggestion
```

**Review by holsoni — CHANGES_REQUESTED** [review]
Two small comments

File: src/main/java/com/polishuchenko/bookstore/config/MapperConfig.java (hunk 1 above)
Comment (holsoni) [reply src/main/java/com/polishuchenko/bookstore/config/MapperConfig.java]: To do it you need to change className to MapperConfiguration

**Review by holsoni — APPROVED** [review]

File: src/main/java/com/polishuchenko/bookstore/repository/BookRepository.java [hunk 3]
```diff
@@ -2,9 +2,12 @@
...
 import com.polishuchenko.bookstore.model.Book;
 import java.util.List;
+import java.util.Optional;
 
 public interface BookRepository {
     Book save(Book book);
 
     List<Book> findAll();
+
+    Optional<Book> findById(Long id);
```
Comment (Vasul-Pankiv) [inline src/main/java/com/polishuchenko/bookstore/repository/BookRepository.java:12]:
```suggestion
    Optional<Book> getById(Long id);
```

File: src/main/java/com/polishuchenko/bookstore/repository/impl/BookRepositoryImpl.java [hunk 4]
```diff
@@ -49,4 +50,13 @@ public List<Book> findAll() {
...
         }
     }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        try (Session session = sessionFactory.openSession()) {
+            return Optional.ofNullable(session.get(Book.class, id));
+        } catch (Exception e) {
+            throw new DataProcessingException("Can't find book with id= " + id, e);
+        }
```
Comment (Vasul-Pankiv) [inline src/main/java/com/polishuchenko/bookstore/repository/impl/BookRepositoryImpl.java:60]: I think its redundant. If you do not get some book you will throw some exception in service

**Review by Vasul-Pankiv — CHANGES_REQUESTED** [review]

**Review by vovasalyha — APPROVED** [review]

**Pull request merged**